include/swift/AST/Types.h

```diff
@@ -2731,6 +2734,8 @@ class ExistentialMetatypeType : public AnyMetatypeType {
   static bool classof(const TypeBase *T) {
     return T->getKind() == TypeKind::ExistentialMetatype;
   }
+
+  Type getExistentialInstanceType();
```

Comment:
I don't like the existence of this method, but it messes up `TypeWalker` if I make `AnyMetatypeType::getInstanceType()` return `ExistentialType` for existential metatypes. Alternatively, I could add `getRawInstanceType()` or something like that for the few places that actually need the stored `InstanceType`. Really, I'd like to just replace `ExistentialMetatypeType` with `ExistentialType(MetatypeType(...))`

Reply:
I think your longer-term goal of removing ExistentialMetatypeType is good and in the meantime, this workaround is reasonable.